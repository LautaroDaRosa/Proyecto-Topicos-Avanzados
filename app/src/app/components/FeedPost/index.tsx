import StFeedPost from './StFeedPost';
import StPusblisher from './StPublisher';
import { Link } from 'react-router-dom';
import ProfileImage from '../ProfileImage';
import StPostPicture from './StPostPicture';
import StDate from './StDate';
import { Post } from 'types';

const getFormattedDate = (date: string) => {
  return date
    ? new Intl.DateTimeFormat('en-US', {
        dateStyle: 'long',
      }).format(new Date(date))
    : '-';
};

const FeedPost = ({ id, content, photo, createdAt, user }: Post) => {
  return (
    <StFeedPost>
      <StPusblisher>
        <ProfileImage type="post-publisher" src={user.photo} alt="User pic" />
        <div>
          <Link to={`/profile/${user.id}`} id={`user-${user.id}`}>
            <span>{user.fullname}</span>
          </Link>
          <StDate>{getFormattedDate(createdAt)}</StDate>
        </div>
      </StPusblisher>
      <span>{content}</span>
      {photo && <StPostPicture src={photo} alt="Post pic!" />}
    </StFeedPost>
  );
};

export default FeedPost;
