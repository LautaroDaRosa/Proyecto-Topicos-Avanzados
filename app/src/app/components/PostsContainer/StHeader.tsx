import styled from 'styled-components';

const StHeader = styled.div`
  display: flex;
  width: 100%;
  padding-bottom: 16px;
  border-bottom: 1px solid #818097;
`;

export default StHeader;
