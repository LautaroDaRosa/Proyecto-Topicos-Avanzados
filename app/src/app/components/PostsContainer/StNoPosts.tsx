import styled from 'styled-components';

const StNoPosts = styled.div`
  display: flex;
  width: 100%;
  gap: 16px;
  flex-direction: column;
  align-items: center;
`;

export default StNoPosts;
