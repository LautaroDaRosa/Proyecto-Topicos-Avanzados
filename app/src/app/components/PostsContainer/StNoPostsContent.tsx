import styled from 'styled-components';

const StNoPostsContent = styled.span`
  font-size: 16px;
  font-style: normal;
  font-weight: 400;
  line-height: normal;
`;

export default StNoPostsContent;
