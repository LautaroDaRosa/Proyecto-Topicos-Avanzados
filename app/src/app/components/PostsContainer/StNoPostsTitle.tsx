import styled from 'styled-components';

const StNoPostsTitle = styled.span`
  font-size: 28px;
  font-style: normal;
  font-weight: 700;
  line-height: normal;
`;

export default StNoPostsTitle;
