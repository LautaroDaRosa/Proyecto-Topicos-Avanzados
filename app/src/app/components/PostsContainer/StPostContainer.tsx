import styled, { css } from 'styled-components';
import { media } from 'styles/media';

const StPostContainer = styled.div<{ hasPosts: boolean }>`
  display: flex;
  flex-direction: column;
  gap: 20px;
  width: 100%;

  ${props =>
    !props.hasPosts &&
    css`
    gap: 24px:

    ${media.medium} {
      gap: 48px;
    }
  
    ${media.large} {
      gap: 60px;
    }
  `}
`;

export default StPostContainer;
