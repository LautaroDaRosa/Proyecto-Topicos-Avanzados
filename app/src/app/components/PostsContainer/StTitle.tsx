import styled from 'styled-components';

const StTitle = styled.span`
  font-size: 20px;
  font-style: normal;
  font-weight: 600;
  line-height: normal;
`;

export default StTitle;
