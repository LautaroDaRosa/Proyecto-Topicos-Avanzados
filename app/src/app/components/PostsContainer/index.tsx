import StHeader from './StHeader';
import StPostContainer from './StPostContainer';
import { ReactNode } from 'react';
import StTitle from './StTitle';
import StNoPostsTitle from './StNoPostsTitle';
import StNoPostsContent from './StNoPostsContent';
import StFlex from '../Flex';

interface PostContainerProps {
  employeeName: string;
  hasPosts: boolean;
  children?: ReactNode;
}

const PostsContainer = ({
  employeeName,
  hasPosts,
  children,
}: PostContainerProps) => {
  return (
    <StPostContainer hasPosts={hasPosts}>
      <StHeader>
        <StTitle>{`${employeeName}'s Posts`}</StTitle>
      </StHeader>
      {hasPosts ? (
        children
      ) : (
        <StFlex
          flexProps={{
            flexDirection: 'column',
            alignItems: 'center',
            gap: 16,
            width: 100,
          }}
        >
          <StNoPostsTitle>Hi there!</StNoPostsTitle>
          <StNoPostsContent>
            {`${employeeName} hasn't posted anything yet. Stay tunned!`}
          </StNoPostsContent>
        </StFlex>
      )}
    </StPostContainer>
  );
};

export default PostsContainer;
