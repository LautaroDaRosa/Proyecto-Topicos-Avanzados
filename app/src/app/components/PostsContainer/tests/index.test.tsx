import { render, screen } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import PostsContainer from '..';
import FeedPost from 'app/components/FeedPost';

describe('Employee Posts', () => {
  it('renders empty posts correctly', () => {
    const name = 'Eliana';
    render(
      <Router>
        <PostsContainer hasPosts={false} employeeName={name} />
      </Router>,
    );

    screen.getByText(name + "'s Posts");
    screen.getByText('Hi there!');
    screen.getByText(name + " hasn't posted anything yet. Stay tunned!");
  });
  it('renders employee posts correctly', () => {
    const post = {
      id: 1,
      content: 'This is my first post!',
      photo: 'image.png',
      key: 1,
      userFullname: 'Eliana Rosselli',
      userPhoto: 'employees/erosselli.png',
      datetime: '2020-07-21T16:38:37.971545Z',
      userId: 12,
    };
    const name = 'Eliana';
    const { container } = render(
      <Router>
        <PostsContainer hasPosts={true} employeeName={name}>
          <FeedPost
            id={post.id}
            content={post.content}
            photo={post.photo}
            key={post.id}
            user={{
              id: post.userId,
              fullname: post.userFullname,
              photo: post.userPhoto,
            }}
            createdAt={post.datetime}
          />
        </PostsContainer>
      </Router>,
    );

    screen.getByText(name + "'s Posts");
    screen.getByText('Eliana Rosselli');
    screen.getByText('This is my first post!');
    screen.getByText('July 21, 2020');
    screen.getByRole('img', { name: 'Post pic!' });
    screen.getByRole('img', { name: 'User pic' });

    expect(container).toMatchSnapshot();
  });
});
