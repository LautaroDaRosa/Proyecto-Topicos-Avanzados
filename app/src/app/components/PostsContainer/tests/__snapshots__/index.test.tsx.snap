// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Employee Posts renders employee posts correctly 1`] = `
.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
  padding-bottom: 16px;
  border-bottom: 1px solid #818097;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  gap: 20px;
  width: 100%;
}

.c2 {
  font-size: 20px;
  font-style: normal;
  font-weight: 600;
  line-height: normal;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  border-radius: 8px;
  background: white;
  box-shadow: 0px 4px 9px 0px rgba(0,0,0,0.1);
  width: 100%;
  padding: 20px;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  gap: 20px;
  font-size: 18px;
  line-height: normal;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  gap: 10px;
}

.c4 div {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c4 a {
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c4 a:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  object-fit: cover;
  height: 48px;
  width: 48px;
  border-radius: 50%;
}

.c7 {
  width: 100%;
}

.c6 {
  color: #7e7e7e;
  font-size: 15px;
  font-style: normal;
  font-weight: 300;
  line-height: normal;
  text-transform: capitalize;
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      <span
        class="c2"
      >
        Eliana's Posts
      </span>
    </div>
    <div
      class="c3"
    >
      <div
        class="c4"
      >
        <img
          alt="User pic"
          class="c5"
          src="employees/erosselli.png"
          type="post-publisher"
        />
        <div>
          <a
            href="/profile/12"
            id="user-12"
          >
            <span>
              Eliana Rosselli
            </span>
          </a>
          <span
            class="c6"
          >
            July 21, 2020
          </span>
        </div>
      </div>
      <span>
        This is my first post!
      </span>
      <img
        alt="Post pic!"
        class="c7"
        src="image.png"
      />
    </div>
  </div>
</div>
`;
